pkg/ccl/streamingccl/streamproducer/event_stream.go

```diff
@@ -361,84 +361,59 @@ func (p *checkpointPacer) shouldCheckpoint(
 	return false
 }
 
-// Add a RangeFeedSSTable into current batch and return number of bytes added.
+// Add a RangeFeedSSTable into current batch.
 func (s *eventStream) addSST(
-	sst *kvpb.RangeFeedSSTable, registeredSpan roachpb.Span, batch *streampb.StreamEvent_Batch,
-) (int, error) {
+	sst *kvpb.RangeFeedSSTable, registeredSpan roachpb.Span, bm *batchManager,
+) error {
 	// We send over the whole SSTable if the sst span is within
 	// the registered span boundaries.
 	if registeredSpan.Contains(sst.Span) {
-		batch.Ssts = append(batch.Ssts, *sst)
-		return sst.Size(), nil
+		bm.addSST(sst)
+		return nil
 	}
 	// If the sst span exceeds boundaries of the watched spans,
 	// we trim the sst data to avoid sending unnecessary data.
 	// TODO(casper): add metrics to track number of SSTs, and number of ssts
 	// that are not inside the boundaries (and possible count+size of kvs in such ssts).
-	size := 0
+	//
 	// Extract the received SST to only contain data within the boundaries of
 	// matching registered span. Execute the specified operations on each MVCC
 	// key value and each MVCCRangeKey value in the trimmed SSTable.
 	if err := replicationutils.ScanSST(sst, registeredSpan,
```

Comment:
Existing code, but if you wanted: `return replicationutils.ScanSST(sst, registeredSpan,...`

Reply:
done.